portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +32,50 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  @Override
+  public void doPost(HttpServletRequest request,HttpServletResponse response) throws IOException {
+      String comment = request.getParameter("text-input");
+      long timestamp = System.currentTimeMillis();
+      if(comment.equals(""))
+      {
+          response.setContentType("text/html;");
+          response.getWriter().println("Please enter a valid comment");
```

Comment:
Please move string literals to class static variables.

`private static final String INVALID_COMMENT_RESPONSE_STRING = "Please enter a valid comment";`